api/schemas/user.py

```diff
@@ -0,0 +1,24 @@
+from typing import Optional
+from pydantic import BaseModel, EmailStr
+
+
+class UserBase(BaseModel):
+    firstname: Optional[str]
+    lastname: Optional[str]
+    email: Optional[EmailStr]
+    password: Optional[str]
+
+
+class UserCreate(BaseModel):
+    firstname: str
+    lastname: str
+    email: EmailStr
+    password: str
+
+
+class UserInfo(UserBase):
```

Comment:
Why do you inherit from UserBase, where all fields are optional, are you going to return not all fields in UserInfo?

Reply:
Yes, I'm going to return different fields for different user roles. Because, common user don't have to see his role and his status(active or not ect).

Reply:
I updated this about the logic of user schemas. Also, I'm not going to return all the fields (`is_active` and `is_admin`). I will return them based on authenticated user permissions